src/main/java/org/fundacionjala/at15/pokemon/Gym.java

```diff
@@ -0,0 +1,34 @@
+package org.fundacionjala.at15.pokemon;
+
+import java.util.*;
+
+public class Gym {
+    private final int hitpoints = 100;
+    private Pokemon gymLeaderPokemon = new Pokemon(hitpoints, "gymLeaderPokemon");
+    private Trainer gymLeader = new Trainer(gymLeaderPokemon, "gymLeader");
```

Comment:
This is only an observation, Trainer should have PokemonTeam instead of Pokemon, so maybe you will have to make a little refactor after that correction.

Reply:
the trainer has a pokemon team, it is constructed with a pokemon team with 1 pokemon by the fault